src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.register.util;
+
+import uk.gov.register.core.HashingAlgorithm;
+
+public class HashValue {
+    private final String rawHash;
+    private final String encodedHash;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String rawHash) {
+        this.rawHash = rawHash;
+        this.encodedHash = hashingAlgorithm.toString() + ":" + rawHash;
+    }
+
+    public String encode() {
+        return encodedHash;
+    }
+
+    public String decode() {
+        return rawHash;
+    }
```

Comment:
Say you have a hash `sha-256:d8630e599d42d13a29836f256a75be7be1be60609d34f5040d3095db23153fbb`. Every time you want to decode this hash, if I were to go by the card in Trello, I'd be returning `new HashValue(rawHash)`. To me this seems wasteful in that we're creating a new `HashValue` object when we will already have an exact duplicate in memory (so long as we're decoding a previously encountered hash).